index.js

```diff
@@ -60,12 +56,49 @@ const gitTasks = opts => {
 	return new Listr(tasks);
 };
 
+const prerequisiteTasks = newVersion => {
+	return new Listr([
+		{
+			title: 'Check Node.js and npm version',
+			skip: () => semver.lt(process.version, '6.0.0'),
+			task: () => execa.stdout('npm', ['version', '--json']).then(json => {
+				const versions = JSON.parse(json);
+				if (!semver.satisfies(versions.npm, '>=2.15.8 <3.0.0 || >=3.10.1')) {
+					return Promise.reject(new Error(`npm@${versions.npm} has known issues publishing when running Node.js 6. Please upgrade npm or downgrade Node and publish again. https://github.com/npm/npm/issues/5082`));
+				}
+			})
+		},
+		{
+			title: 'Fetch remote changes',
+			task: () => execa('git', ['fetch'])
+		},
+		{
+			title: 'Check git tag existence',
+			task: () => execa.stdout('git', ['rev-parse', '--quiet', '--verify', `refs/tags/v${newVersion}`])
+				.then(
+					output => {
+						if (output) {
+							throw new Error(`Git tag \`v${newVersion}\` already exists.`);
+						}
+					},
+					err => {
+						if (err.stdout !== '' || err.stderr !== '') {
+							throw err;
+						}
+					}
+				)
+		}
+	]);
+};
+
 module.exports = (input, opts) => {
 	input = input || 'patch';
 	opts = opts || {};
 
 	const runTests = !opts.yolo;
 	const runCleanup = !opts.skipCleanup && !opts.yolo;
+	const pkg = readPkgUp.sync().pkg;
+	const newVersion = semver.inc(pkg.version, input);
```

Comment:
This doesn't work if the input is a specific version that's lower than the current version. For example, the package version is `0.4.0` and you do `np 0.3.0`. `newVersion` is then `null` and it passes all the prerequisite tests. Maybe we should have a separate check about the version being higher than the current?